### PR TITLE
ShuffleChannels ng op and reference implementation revision

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
Revision of ShuffleChannels ng op and reference implementation.

- [x] Reference implementation update (refactoring, ND input support)
- [x] Preserve output shape dynamic bounds
- [x] Check RTTI_DECLARATION macro (already updated)
- [x] Backend test (moved from fused_op tests and update)
- [x] Update Type_prop tests
- [x] Attribute visitor test
- [x] Add single layer tests

There is a restriction in the [spec ](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_ShuffleChannels_1.html) that data input has to be 4D, but v0::ShuffleChannels ng op allows (and allowed before) input shapes 1D and bigger what is also supported by CPU. The spec is going to be updated with such info by PR https://github.com/openvinotoolkit/openvino/pull/5813.

### Tickets:
 - 44622 (45791, 45792, 45793, 53946)

### CODE REVIEW

**Review by chenhu-wang — APPROVED** [review]

**Comment by chenhu-wang** [comment]
evaluate() looks good and is aligned with CPU plugin.

File: ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cfenv>
```
Comment (pelszkow) [inline ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp:7]: Is this include used?

File: ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include "ngraph/runtime/opt_kernel/reshape.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            void shuffle_channels(const char* arg,
```
Comment (pelszkow) [inline ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp:21]: Do you consider to move this impl to cpp file?

**Review by pelszkow — APPROVED** [review]

**Comment by jdanieck** [comment]
@tomdol maybe you'll have time to review & merge to offload @ilyachur ? Unreviewed PR's started to pile up because of the release. CC: @andrei-kochin

File: ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp (hunk 2 above)
Comment (mitruska) [reply ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp:19]: Considered and done

File: ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp (hunk 1 above)
Comment (mitruska) [reply ngraph/core/reference/include/ngraph/runtime/reference/shuffle_channels.hpp]: I don't think so, removed.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**